src/mlpack/tests/test_tools.hpp

```diff
@@ -171,5 +171,39 @@ inline std::string FilterFileName(const std::string& inputString)
 
   return fileName;
 }
+//Templated Check for 2 matrices of type nested vectors
+template<typename N>
+inline void CheckMatrices( std::vector<std::vector<N>> vec1, std::vector<std::vector<N>> vec2, float tolerance=1e-3)
+{
 
+  BOOST_REQUIRE_EQUAL( vec1.size() , vec2.size() );
+  for ( size_t i = 0; i < vec1.size() ; i++)
+  {
+    BOOST_REQUIRE_EQUAL( vec1[i].size(), vec2[i].size() );
+    std::sort(vec1[i].begin(), vec1[i].end());
+    std::sort(vec2[i].begin(), vec2[i].end());
+    for (size_t j = 0 ; j < vec1[i].size() ; j++)
+    {
+      BOOST_REQUIRE_CLOSE(static_cast<float>(vec1[i][j]), static_cast<float>(vec2[i][j]), tolerance);
```

Comment:
Casting every element to a `float` sort of breaks the purpose of the template parameter here.  Since `size_t` needs to be compared with `BOOST_REQUIRE_EQUAL()` and `double` needs to be compared with `BOOST_REQUIRE_CLOSE()` I would suggest making two overloads specific to those cases.

It's up to you, but I'd also consider putting this at the top of `range_search_test.cpp` since it is only used there.